pkg/cmd/cli/cmd/observe/observe.go

```diff
@@ -0,0 +1,936 @@
+package observe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"text/template"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/healthz"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/jsonpath"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/util/proc"
+)
+
+var (
+	observeCounts = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "observe_counts",
+			Help: "Number of changes observed to the underlying resource.",
+		},
+		[]string{"type"},
+	)
+	execDurations = prometheus.NewSummaryVec(
```

Comment:
Hrm - if we had app metrics that would be captured.  I can kind of buy
printing them to expose errors.

On Tue, Aug 30, 2016 at 2:31 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/observe/observe.go
> <https://github.com/openshift/origin/pull/4196#discussion_r76852957>:
>
> > +	"k8s.io/kubernetes/pkg/util/jsonpath"
> > +	"k8s.io/kubernetes/pkg/watch"
> > +
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	"github.com/openshift/origin/pkg/util/proc"
> > +)
> > +
> > +var (
> > +	observeCounts = prometheus.NewCounterVec(
> > +		prometheus.CounterOpts{
> > +			Name: "observe_counts",
> > +			Help: "Number of changes observed to the underlying resource.",
> > +		},
> > +		[]string{"type"},
> > +	)
> > +	execDurations = prometheus.NewSummaryVec(
>
> Glad to have these to begin with.
>
> Seems like these should be dumped on a crash (non-zero exit code), since
> they are irretrievable at that point.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/4196/files/69ab95173b3a274314ae59c42690a33690f9f69e#r76852957>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p6E78xf_xpwFDzmudZ-lN52MSbc6ks5qlHcbgaJpZM4Fsaph>
> .
>